### PR TITLE
Update principles doc

### PR DESCRIPTION
**Pull request by mthalman**
There have been recent discussions (https://github.com/dotnet/dotnet-docker/pull/4822#discussion_r1302052550, https://github.com/dotnet/dotnet-docker/pull/4822#discussion_r1302102256) on which packages should be listed for installation in our Dockerfiles. These have been resolved and captured in the principles doc.

### CODE REVIEW

File: documentation/guiding-principles.md [hunk 1]
```diff
@@ -29,6 +29,11 @@ These are the guiding principles for the content, tagging and production of the
 
 1. The matrix of [supported container operating systems and versions](supported-platforms.md) will evolve on a continuous basis.
 
+1. For packages that are installed from a native package manager:
+    * When a package is installed, it will be done so in a way that forces it to be upgraded. This ensures the latest version of the package is installed even if an older version of the package was installed in a base image.
+    * A package will be listed for install if it is a top-level packages necessary to fulfill required scenarios for the image.
```
Comment (lbussell) [inline documentation/guiding-principles.md:34]:
```suggestion
    * A package will be listed for install if it is a top-level package necessary to fulfill required scenarios for the image.
```

File: documentation/guiding-principles.md [hunk 2]
```diff
@@ -29,6 +29,11 @@ These are the guiding principles for the content, tagging and production of the
 
 1. The matrix of [supported container operating systems and versions](supported-platforms.md) will evolve on a continuous basis.
 
+1. For packages that are installed from a native package manager:
+    * When a package is installed, it will be done so in a way that forces it to be upgraded. This ensures the latest version of the package is installed even if an older version of the package was installed in a base image.
+    * A package will be listed for install if it is a top-level package necessary to fulfill required scenarios for the image.
+    * A dependency of a top-level package will not be listed for install unless it is deemed to have sufficient CVE activity that would make it beneficial to do so.
```
Comment (MichaelSimons) [inline documentation/guiding-principles.md:35]: Is it true at least for Alpine that it doesn't matter if dependencies are listed or not, apk add --upgrade will upgrade any dependencies to latest.  Can this be achieved on all distros?  It seems broken that we would have to list the dependencies.

File: documentation/guiding-principles.md [hunk 3]
```diff
@@ -29,6 +29,11 @@ These are the guiding principles for the content, tagging and production of the
 
 1. The matrix of [supported container operating systems and versions](supported-platforms.md) will evolve on a continuous basis.
 
+1. For packages that are installed from a native package manager:
+    * When a package is installed, it will be done so in a way that forces it to be upgraded. This ensures the latest version of the package is installed even if an older version of the package was installed in a base image.
```
[the remark this answers is not in the document]
Comment (mthalman) [reply documentation/guiding-principles.md:33]: This has always been the intended policy. Alpine was the only one that didn't have this behavior until we discovered it and fixed it: https://github.com/dotnet/dotnet-docker/issues/4776.

[the remark this answers is not in the document]
Comment (MichaelSimons) [reply documentation/guiding-principles.md:33]: Thanks for the clarification/reminder.

File: documentation/guiding-principles.md (hunk 2 above)
Comment (mthalman) [reply documentation/guiding-principles.md:35]: > Is it true at least for Alpine that it doesn't matter if dependencies are listed or not, apk add --upgrade will upgrade any dependencies to latest.

Yes

> Can this be achieved on all distros? It seems broken that we would have to list the dependencies.

Not without a more complicated implementation. `apt-get` has a [`--only-upgrade` option](https://manpages.debian.org/bullseye/apt/apt-get.8.en.html) but that only upgrades the listed package, not dependencies. To really implement the behavior we'd want, you need a script which enumerates the package hierarchy and use the `--only-upgrade` option for them all.

**Comment by mthalman** [comment]
I'm going to close this for now. Now that I look at it, this content doesn't belong in the principles doc. It's more of an image support policy thing which are covered by several other doc issues, notably https://github.com/dotnet/dotnet-docker/issues/2787.

**Pull request closed without merging**